core/qthelper.cpp

```diff
@@ -346,19 +345,64 @@ extern "C" xsltStylesheetPtr get_stylesheet(const char *name)
 	return xslt;
 }
 
+// Fetch quint16 in big endian mode from QFile and return 0 on error.
+// This is a very specialized function for parsing JPEGs, therefore we can get away with such an in-band error code.
+static inline quint16 getShortBE(QFile &f)
```

Comment:
Wouldn't be these two functions better be placed in exif.cpp?

Reply:
Could be. I didn't want to touch an external BSD-licensed thing. That would make merging updates to the EXIF code more difficult. I was thinking about creating a new `core/metadata.cpp` file, once we can also parse mp4 metadata, etc. But that's dreaming of the future.

Reply:
We already slightly modified that file. But maybe a separate one would be a good idea (ideally placing our other changes there as well).

Reply:
@dirkhh: ACK for creating a new `core/metadata.cpp` file containing the above EXIF helper functions and MP4 parsing? According to a quick internet search, extracting media creation date from MP4s seems to be rather trivial.

Reply:
I'm ok with creating a metadata.cpp file